src/components/App.tsx

```diff
@@ -1,21 +1,62 @@
-import { lazy } from 'react';
+import { createContext, lazy, useEffect, useState } from 'react';
 import { Route, Routes } from 'react-router-dom';
 import AppBar from './appBar/AppBar';
 import Home from './pages/home/Home';
 
 const Details = lazy(() => import('./pages/details/Details'));
 const Garden = lazy(() => import('./pages/garden/Garden'));
 
+export type DataItem = {
```

Comment:
types and interfaces better keep in a separate folder in src/types or src/interfaces
if you are defining the type for an object its better to use interface